compiler/frontend/src/module/module.rs

```diff
@@ -10,29 +10,59 @@ use std::{
     fs,
     hash::Hash,
     path::{Path, PathBuf},
+    sync::Arc,
 };
 use tracing::{error, warn};
 
 #[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
-pub struct Module {
-    pub package: Package,
-    pub path: Vec<String>,
-    pub kind: ModuleKind,
+pub struct Module(Arc<InnerModule>);
+#[derive(Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
+struct InnerModule {
+    package: Package,
+    path: Vec<String>,
+    kind: ModuleKind,
 }
 #[derive(Clone, Copy, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub enum ModuleKind {
     Code,
     Asset,
 }
-
+impl From<InnerModule> for Module {
+    fn from(val: InnerModule) -> Self {
+        Self(Arc::new(val))
+    }
+}
 impl Module {
+    #[must_use]
+    pub fn package(&self) -> &Package {
+        &self.0.package
+    }
+    #[must_use]
+    pub fn path(&self) -> &Vec<String> {
+        &self.0.path
+    }
+    #[must_use]
+    pub fn kind(&self) -> ModuleKind {
+        self.0.kind
+    }
```

Comment:
These accessors should go below the “constructors” (`new(…)`, `from_path(…)`, …)